### PR TITLE
Tenant branding document and changelog

### PR DESCRIPTION
**Pull request by anchandaAADGraph**

### CODE REVIEW

File: api-reference/beta/api/organization-list-brandings.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+author: "anchanda"
+ms.prod: "branding"
+doc_type: "apiPageType"
+---
+
+# List brandings
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Retrieve a list of organizationalbranding objects.
```
Comment (dkershaw10) [inline api-reference/beta/api/organization-list-brandings.md]: organizationalbranding -> organizationalBranding

File: api-reference/beta/api/organization-list-brandings.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+## Permissions
+
+One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).
+
+| Permission type                        | Permissions (from least to most privileged) |
+|:---------------------------------------|:--------------------------------------------|
+| Delegated (work or school account)     | orgContacts.Read.All |
```
Comment (dkershaw10) [inline api-reference/beta/api/organization-list-brandings.md]: These are wrong.  Should be `Organization.Read.All`, no?

File: api-reference/beta/api/organization-list-brandings.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+| Delegated (personal Microsoft account) | orgContacts.Read.All |
```
Comment (dkershaw10) [inline api-reference/beta/api/organization-list-brandings.md]: Should say Not supported

File: api-reference/beta/api/organization-list-brandings.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+## Optional query parameters
+
+This method supports some of the OData query parameters to help customize the response. For general information, see [OData Query Parameters](/graph/query-parameters)
+
+## Request headers
+
+| Name      |Description|
+|:----------|:----------|
+| Authorization | Bearer {code} |
```
Comment (dkershaw10) [inline api-reference/beta/api/organization-list-brandings.md]: did this come from the stub generator tool?  I think this should be token, not code.

File: api-reference/beta/api/organization-list-brandings.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
+---
+title: "List brandings"
+description: "Retrieve a list of organizationalbranding objects."
+localization_priority: Normal
+author: "anchanda"
+ms.prod: "branding"
+doc_type: "apiPageType"
+---
+
+# List brandings
```
Comment (dkershaw10) [inline api-reference/beta/api/organization-list-brandings.md]: brandings?
The API spec called this "branding".  It should be branding.  Please confirm.  

Title should be called "List organizationalBranding"

File: api-reference/beta/api/organization-post-brandings.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+Use this API to create a new organizationalBranding.
+
+## Permissions
+
+One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).
+
+| Permission type                        | Permissions (from least to most privileged) |
+|:---------------------------------------|:--------------------------------------------|
+| Delegated (work or school account)     | Organization.ReadWrite.All |
+| Delegated (personal Microsoft account) | Organization.ReadWrite.All |
```
Comment (dkershaw10) [inline api-reference/beta/api/organization-post-brandings.md]: Not supported.

File: api-reference/beta/api/organization-list-brandings.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
+---
+title: "List brandings"
```
Comment (dkershaw10) [inline api-reference/beta/api/organization-list-brandings.md]: You are missing:
1. A get topic
2. In the get topic you need to provide details on getting the default and a specific language branding, and how this works.

File: api-reference/beta/resources/organizationalbranding.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+author: "anchanda"
+ms.prod: "branding"
+doc_type: "resourcePageType"
+---
+
+# organizationalBranding resource type
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+PROVIDE DESCRIPTION HERE
```
Comment (dkershaw10) [inline api-reference/beta/resources/organizationalbranding.md]: Pleasew fill in with details.  What scenarios this covers, what it does, link to the topic on tenant branding that already exists here: https://docs.microsoft.com/en-us/azure/active-directory/fundamentals/customize-branding.  Also call out the license/subscription requirements
You should also describe that the materials can be localized, so please detail how this works at a high level.  The API topics should cover examples of this.

File: api-reference/beta/api/organization-list-brandings.md (hunk 2 above)
Comment (anchandaAADGraph) [reply api-reference/beta/api/organization-list-brandings.md]: Oh yes.

File: api-reference/beta/api/organization-list-brandings.md (hunk 4 above)
Comment (anchandaAADGraph) [reply api-reference/beta/api/organization-list-brandings.md]: Yes it's from the stub tool. Shall I change it?

File: api-reference/beta/api/organization-list-brandings.md (hunk 5 above)
Comment (anchandaAADGraph) [reply api-reference/beta/api/organization-list-brandings.md]: We've exposed it as brandings. Analogous to users, groups etc. Single object is organizational branding. Changing the title.

File: concepts/changelog.md [hunk 9]
```diff
@@ -26,6 +26,12 @@ For details about known issues with Microsoft Graph APIs, see [Known issues](kno
 | :-------------- | :------------ | :--------------------------------------- |
 | Addition | beta | Added the [teamDiscoverySettings](/api-reference/beta/resources/teamdiscoverysettings.md) resource and associated methods. |
 
+### Directory APIs
+| **Change type** | **Version** | **Description** |
+| :-------------- | :------------ | :--------------------------------------- |
+| Addition | beta | Added **brandings** property to the [organization](https://docs.microsoft.com/en-us/graph/api/resources/organization?view=graph-rest-beta) entity. This property represents the list of organizational branding elemnets as defined by tenant.|
```
Comment (dkershaw10) [inline concepts/changelog.md]: elemnets -> elements

File: api-reference/beta/api/organization-list-brandings.md (hunk 7 above)
Comment (anchandaAADGraph) [reply api-reference/beta/api/organization-list-brandings.md]: adding

File: api-reference/beta/api/organization-post-brandings.md (hunk 6 above)
Comment (anchandaAADGraph) [reply api-reference/beta/api/organization-post-brandings.md]: Done.

File: api-reference/beta/resources/organizationalbranding.md (hunk 8 above)
Comment (anchandaAADGraph) [reply api-reference/beta/resources/organizationalbranding.md]: Sure

File: api-reference/beta/api/organization-list-brandings.md (hunk 5 above)
Comment (Lauragra) [reply api-reference/beta/api/organization-list-brandings.md]: @anchandaAADGraph , do you plan to update the title?

File: api-reference/beta/api/organization-list-brandings.md [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+author: "anchanda"
+ms.prod: "branding"
+doc_type: "apiPageType"
+---
+
+# List brandings
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Retrieve a list of organizationalBranding objects.
```
Comment (Lauragra) [inline api-reference/beta/api/organization-list-brandings.md]: Please make the first mention of the resource name a link.

File: api-reference/beta/api/organization-list-brandings.md [hunk 11]
```diff
@@ -0,0 +1,154 @@
...
+
+# Get branding
```
Comment (Lauragra) [inline api-reference/beta/api/organization-list-brandings.md]: This looks like an error - a topic can't have more than one H1 (#). Please remove.

File: api-reference/beta/api/organization-list-brandings.md [hunk 12]
```diff
@@ -0,0 +1,154 @@
...
+
+<!-- { "blockType": "ignored" } -->
+
+List of all brandings
+
+```http
+GET /organization/{id}/brandings
+```
+Get Branding from a locale
+
```
Comment (Lauragra) [inline api-reference/beta/api/organization-list-brandings.md]: Are we trying to combine list and get operations in a single API topic? If both operations are supported, we should have separate API topics for each. So, create a separate topic for Get organizationBranding, and include the description and syntax for GET in that topic.

Comment (dkershaw10) [reply api-reference/beta/api/organization-list-brandings.md]: @anchandaAADGraph We only support GET an item for this API, and not list AFAIK.  Please make sure you fix this topic accordingly AND rename the file, titles etc.

File: api-reference/beta/api/organization-list-brandings.md [hunk 13]
```diff
@@ -0,0 +1,154 @@
...
+```http
+GET /organization/{id}/brandings/{locale}
```
Comment (dkershaw10) [inline api-reference/beta/api/organization-list-brandings.md]: What format is the locale in? How do I find out???

File: api-reference/beta/api/organization-list-brandings.md [hunk 14]
```diff
@@ -0,0 +1,154 @@
...
+```
+
+## Optional query parameters
+
+This method supports some of the OData query parameters to help customize the response. For general information, see [OData Query Parameters](/graph/query-parameters)
```
Comment (dkershaw10) [inline api-reference/beta/api/organization-list-brandings.md]: Does it really support any of these?  Please clarify.

File: api-reference/beta/api/organization-list-brandings.md [hunk 15]
```diff
@@ -0,0 +1,154 @@
...
+### Request
+
+The following is an example of the branding request.
+<!-- {
+  "blockType": "request",
+  "name": "get_branding"
+}-->
+
+```http
+GET https://graph.microsoft.com/beta/organization/{id}/brandings
```
Comment (dkershaw10) [inline api-reference/beta/api/organization-list-brandings.md]: If this is supported, we need a separate get and list topic.

File: api-reference/beta/api/organization-post-brandings.md [hunk 16]
```diff
@@ -0,0 +1,170 @@
...
+author: "anchanda"
+ms.prod: "branding"
+doc_type: "apiPageType"
+---
+
+# Create organizationalBranding
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Use this API to create a new organizationalBranding.
```
Comment (dkershaw10) [inline api-reference/beta/api/organization-post-brandings.md]: Link to the type definition.

File: api-reference/beta/api/organization-post-brandings.md [hunk 17]
```diff
@@ -0,0 +1,170 @@
...
+  "id": "id-value",
+  "backgroundColor": "backgroundColor-value",
+  "backgroundImage": "backgroundImage-value",
+  "bannerLogo": "bannerLogo-value",
+  "locale": "locale-value",
+  "signInPageText": "signInPageText-value"
+}
+```
+
+### Request
```
Comment (dkershaw10) [inline api-reference/beta/api/organization-post-brandings.md]: You need to create a **separate topic file for update**.  And I thought that used a PUT operation, not POST

File: api-reference/beta/resources/organizationalbranding.md [hunk 18]
```diff
@@ -0,0 +1,83 @@
...
+author: "anchanda"
+ms.prod: "branding"
+doc_type: "resourcePageType"
+---
+
+# organizationalBranding resource type
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Organizations can customize their Azure AD sign-in pages, which appear when users sign in to the organization's tenant-specific apps, such as https://outlook.com/contoso.com , or when passing a domain variable, such as https://passwordreset.microsoftonline.com/?whr=contoso.com . Also as an ISV developer, they can read the company's branding information, and customize their app experience and tailor it specifically for the signed-in user, using their company's [branding](https://docs.microsoft.com/en-us/azure/active-directory/fundamentals/customize-branding).
```
Comment (dkershaw10) [inline api-reference/beta/resources/organizationalbranding.md]: This doesn't meet the style guide.
Please rewrite so it's more personal.  You can use this API to read a company's branding information to customize your app's experience specifically for your signed-in users.

File: api-reference/beta/resources/organizationalbranding.md [hunk 19]
```diff
@@ -0,0 +1,83 @@
...
+
+Companies can add different multiple branding based on locale. Locale serves as a key in all requests.
```
Comment (dkershaw10) [inline api-reference/beta/resources/organizationalbranding.md]: different multiple???
Locale serves as a key in all requests.  - not true if you support List...

File: api-reference/beta/resources/organizationalbranding.md [hunk 20]
```diff
@@ -0,0 +1,83 @@
...
+NOTE: The custom branding won't immediately appear when users go to sites such as, www.office.com . Instead, users have to enter their username before their customized branding appears.
+
+> [!NOTE]
+> Adding custom branding requires you to use Azure Active Directory Premium 1, Premium 2, or Basic editions, or to have an Office 365 license. For more information about licensing and editions, see Sign up for Azure AD Premium.
+
+## Methods
+
+| Method       | Return Type | Description |
+|:-------------|:------------|:------------|
+| [Get organizationalBranding](../api/organizationalbranding-get.md) | [organizationalBranding](organizationalbranding.md) | Read properties and relationships of organizationalBranding object. |
```
Comment (dkershaw10) [inline api-reference/beta/resources/organizationalbranding.md]: Make sure this list matches your set of API method topics.

File: api-reference/beta/resources/organizationalbranding.md [hunk 21]
```diff
@@ -0,0 +1,83 @@
...
+|backgroundColor|String|Color that will appear in place of the background image in low-bandwidth connections. The primary color of your banner logo or your organization color is recommended to be used here. Specify this in hexadecimal (for example, white is #FFFFFF).|
+|backgroundImage|Stream|Image that appears as the background of the sign in page. .png or .jpg not larger than 1920x1080 and smaller than 300kb.|
+|bannerLogo|Stream|A banner version of your company logo which appears appears on the sign-in page. .png or .jpg no larger than 36x245px. We recommend using a transparent image with no padding around the logo.|
+|id|String| Id is same as locale property for now. Adding this due to be complient to MSGraph inheritance property. Going forward if we choose to expose functionality to have multiple brandings for one locale, this can be changed to hold different value. '0' for Default.|
+|locale|String|Locale serves as Id for the branding. It specifies the language and ISO-639 string value (eg. en-US for English-US). Tenant can have multiple brandings based on locale. The language is automatically set as your default language, set while tenant creation:'0'. Read-only for PATCH|
```
Comment (dkershaw10) [inline api-reference/beta/resources/organizationalbranding.md]: Id -> ID
It specifies the language and ISO-639 string value (eg. en-US for English-US). Tenant can have multiple brandings based on locale.  -> It specifies the language using ISO-639 (for example en-US for English-US). An organization can have multiple branding definitions per locale. 
This sentence makes absolutely no sense - please fix: "The language is automatically set as your default language, set while tenant creation:'0'."   And this is not ISO-639, so you need to explain that away.
Read-only for PATCH -> Read-only for updates.

File: api-reference/beta/api/organization-list-brandings.md (hunk 5 above)
Comment (anchandaAADGraph) [reply api-reference/beta/api/organization-list-brandings.md]: No. I would prefer keeping it the same.

File: api-reference/beta/api/organization-list-brandings.md (hunk 10 above)
Comment (anchandaAADGraph) [reply api-reference/beta/api/organization-list-brandings.md]: sure!

File: api-reference/beta/api/organization-list-brandings.md (hunk 12 above)
Comment (anchandaAADGraph) [reply api-reference/beta/api/organization-list-brandings.md]: We do support list. I can add separate doc for it. I assumed both are GET calls so should come under same topic.

File: api-reference/beta/api/organization-list-brandings.md (hunk 15 above)
Comment (anchandaAADGraph) [reply api-reference/beta/api/organization-list-brandings.md]: sure.

File: api-reference/beta/api/organization-post-brandings.md (hunk 16 above)
Comment (anchandaAADGraph) [reply api-reference/beta/api/organization-post-brandings.md]: done.

File: api-reference/beta/api/organization-post-brandings.md (hunk 17 above)
Comment (anchandaAADGraph) [reply api-reference/beta/api/organization-post-brandings.md]: on it,

File: api-reference/beta/resources/organizationalbranding.md (hunk 20 above)
Comment (anchandaAADGraph) [reply api-reference/beta/resources/organizationalbranding.md]: yep resolving.

File: api-reference/beta/resources/organizationalbranding.md (hunk 21 above)
Comment (anchandaAADGraph) [reply api-reference/beta/resources/organizationalbranding.md]: Resolved

File: api-reference/beta/api/organization-list-brandings.md (hunk 13 above)
Comment (anchandaAADGraph) [reply api-reference/beta/api/organization-list-brandings.md]: We're explaining that in organizationbranding object. But i'll add it here as well.

File: api-reference/beta/api/organization-post-brandings.md (hunk 17 above)
Comment (anchandaAADGraph) [reply api-reference/beta/api/organization-post-brandings.md]: >PUT is for images. POST for branding creation. 
>Adding all topics.

**Comment by Lauragra** [comment]
@anchandaAADGraph , it looks like the API is live? What is the status of this PR? Have you incorporated Dan's feedback?

**Comment by Lauragra** [comment]
@anchandaAADGraph , this API is live in beta. I see that comments are resolved but I don't see the updates in the files. Did you commit your changes? Can you please address all feedback so we can get this ready to publish? @dkershaw10 , FYI.

**Comment by Lauragra** [comment]
Closing, @kelvinxia-MSFT is creating a new set of docs for this feature.

**Pull request closed without merging**